tests_integ/helpers/base_test.py

```diff
@@ -0,0 +1,345 @@
+
+        if not resources:
+            return []
+
+        return self.api_client.get_stages(restApiId=resources[0]["PhysicalResourceId"])["item"]
+
+    def get_stack_stages_v2(self):
+        resources = self.get_stack_resources("AWS::ApiGatewayV2::Api")
+
+        if not resources:
+            return None
+
+        return self.api_v2_client.get_stages(ApiId=resources[0]["PhysicalResourceId"])["Items"]
+
+    def get_stack_nested_stack_resources(self):
+        resources = self.get_stack_resources("AWS::CloudFormation::Stack")
+
+        if not resources:
+            return None
+
+        return self.cloudformation_client.list_stack_resources(StackName=resources[0]["PhysicalResourceId"])
+
+    def get_stack_outputs(self):
+        if not self.stack_description:
+            return {}
+        else:
+            output_key_to_value = {}
+            output_list = self.stack_description["Stacks"][0]["Outputs"]
+            for output in output_list:
+                output_key_to_value[output["OutputKey"]] = output["OutputValue"]
+            return output_key_to_value
+
+    def get_resource_status_by_logical_id(self, logical_id):
+        if not self.stack_resources:
+            return ""
+
+        for res in self.stack_resources["StackResourceSummaries"]:
+            if res["LogicalResourceId"] == logical_id:
+                return res["ResourceStatus"]
+
+        return ""
+
+    def get_physical_id_by_type(self, resource_type):
+        if not self.stack_resources:
+            return ""
+
+        for res in self.stack_resources["StackResourceSummaries"]:
+            if res["ResourceType"] == resource_type:
+                return res["PhysicalResourceId"]
+
+        return ""
+
+    def get_logical_id_by_type(self, resource_type):
+        if not self.stack_resources:
+            return ""
+
+        for res in self.stack_resources["StackResourceSummaries"]:
+            if res["ResourceType"] == resource_type:
+                return res["LogicalResourceId"]
+
+        return ""
+
+    def get_physical_id_by_logical_id(self, logical_id):
+        if not self.stack_resources:
+            return ""
+
+        for res in self.stack_resources["StackResourceSummaries"]:
+            if res["LogicalResourceId"] == logical_id:
+                return res["PhysicalResourceId"]
+
+        return ""
+
+    def _fill_template(self, file_name):
+        """
+        Replaces the template variables with their value
+
+        Parameters
+        ----------
+        file_name : string
+            Template file name
+        """
+        input_file_path = str(Path(self.template_dir, file_name + ".yaml"))
+        updated_template_path = str(Path(self.output_dir, "sub_" + file_name + ".yaml"))
+        with open(input_file_path, "r") as f:
+            data = f.read()
+        for key, _ in CODE_KEY_TO_FILE_MAP.items():
+            data = data.replace("${{{}}}".format(key), self.get_code_key_s3_uri(key))
```

Comment:
> `${{{}}}` 

What's going on here?

Reply:
The end result we need is "${mykeyvalue}", I'll add a comment